### ISSUE TITLE
yii2-multiple-input + yii2-fileapi-widget = bug

### ISSUE THREAD

**Issue by kotchuprik** [unclead/yii2-multiple-input#12]
Thanks for the cool extension. I'm using [yii2-fileapi-widget extension](https://github.com/vova07/yii2-fileapi-widget) and I have an error: 

![Alt text](https://monosnap.com/file/hNv7eXziIQMFMueYHahVwQ9Q5LYhHq.png)

I tried to fix it, but I did not succeed. Could you to advise a solution?

**Comment by unclead** [comment]
@kotchuprik could you provide code example for reproducing this bug? I'll try to fix it

**Comment by kotchuprik** [comment]
sure

```
<?php $form = \yii\bootstrap\ActiveForm::begin([
    'enableAjaxValidation' => true,
    'enableClientValidation' => false,
    'validateOnChange' => false,
    'validateOnSubmit' => true,
    'validateOnBlur' => false,
]) ?>

<?= $form->field($model, 'items')->label(false)->widget(\unclead\widgets\MultipleInput::className(), [
    'columns' => [
        [
            'name' => 'title',
            'title' => $model->getAttributeLabel('title'),
            'type' => \unclead\widgets\MultipleInputColumn::TYPE_TEXT_INPUT,
        ],
        [
            'name' => 'description',
            'title' => $model->getAttributeLabel('description'),
            'type' => \vova07\imperavi\Widget::className(),
        ],
        [
            'name' => 'file',
            'title' => $model->getAttributeLabel('file'),
            'type' => \vova07\fileapi\Widget::className(),
            'options' => [
                'model' => $model,
                'attribute' => 'items',
            ],
        ],
    ],
]) ?>

<div class="form-group">
    <?= \yii\helpers\Html::submitButton('Сохранить', ['class' => 'btn btn-primary']) ?>
</div>

<?php \yii\bootstrap\ActiveForm::end() ?>
```

But it's not right code :) I set `model` and `items`. With this code you could see js error. If I remove `model` and `attribute` (right way) will be an error.

**Comment by unclead** [comment]
Yes, FileApi widget can not works without model and attribute, i realized it yesterday when tried to reproduce issue.

I think we can add comment to doc about this feature because fix it on the widget's side is not right way. Its impossible cover all kind of external widgets

**Comment by kotchuprik** [comment]
good. what about js error? how we could specify `model` and `attribute` from cycle? I mean `file` attribute from `items` array.

**Comment by unclead** [comment]
I need time to think about it. Unfortunately i don't have enough time now, too busy on my work.
I'll try to fix it during the week

**Comment by kotchuprik** [comment]
thanks. I will think too, because I need something like fileapi widget :)

**Comment by unclead** [comment]
I think this case looks like tabular input and attributes title, description and file wont stored in an attribute of model.

In other words, you have model with attributes title, description and file and you want to use my widgets to handle tabular input

Am i right?

**Comment by kotchuprik** [comment]
yes. updating some models via tabular input.

**Comment by unclead** [comment]
Nowadays i think that it is impossible to use my widget in case of handling tabular input without without serious improvements. I can make `TabularInput` widget with similar functionality with Multiple Input, but there is another good solution for this case. For example https://github.com/kartik-v/yii2-builder

MultipleInput has different conception

I would like to implement such functionality, but i'm too busy now om my main job. If you able to implement this by yourself it would be great.

I will leave this issue opened

**Comment by unclead** [comment]
depends on #22

**Comment by unclead** [comment]
@kotchuprik finally i made it :smile: 
But i cannot understand how yii2-fileapi-widget works. I submit еру form after i selected the file, but i'm getting an empty result on server side...

**Comment by unclead** [comment]
closed via 3083574758d18a6870f0c44b49370c2b886b1d28
